src/apps/tools.clj

```diff
@@ -96,19 +108,24 @@
   [{:keys [tools]}]
   (transaction
     (let [tool-ids (doall (map add-new-tool tools))]
+      (dorun (map permissions/register-public-tool tool-ids))
       {:tool_ids tool-ids})))
 
 (defn update-tool
-  [overwrite-public {:keys [id container] :as tool}]
+  [user overwrite-public {:keys [id container] :as tool}]
   (persistence/update-tool tool)
   (when container
     (set-tool-container id overwrite-public container))
-  (get-tool id))
+  (get-tool user id))
```

Comment:
I don't see where this function is checking tool permissions before allowing the update.